### PR TITLE
Add Dogma and TravisCI

### PR DESCRIPTION
**Pull request by doomspork**
RFC @hassox / @scrogson — Like the other PRs this adds Dogma and TravisCI to the example project.

### CODE REVIEW

**Comment by doomspork** [comment]
Do we want to add a TravisCI badge to the README?

File: mix.lock [hunk 1]
```diff
@@ -25,7 +26,7 @@
   "ranch": {:hex, :ranch, "1.2.1"},
   "ssl_verify_hostname": {:hex, :ssl_verify_hostname, "1.0.5"},
   "ueberauth": {:hex, :ueberauth, "0.2.0"},
-  "ueberauth_facebook": {:hex, :ueberauth_facebook, "0.2.1"},
+  "ueberauth_facebook": {:git, "https://github.com/ueberauth/ueberauth_facebook.git", "c59ef2273539f485a5f1c44149aeb6c7a3c0e526", [branch: "support-auth-type"]},
```
Comment (doomspork) [inline mix.lock]: I need to fix this.

File: web/models/user_from_auth.ex [hunk 2]
```diff
@@ -20,12 +28,22 @@ defmodule UserFromAuth do
...
   end
 
-  defp validate_password(%Credentials{other: %{password: ""}}), do: {:error, "Password required"}
-  defp validate_password(%Credentials{other: %{password: pw, password_confirmation: pw}}), do: :ok
-  defp validate_password(%Credentials{other: %{password: _}}), do: { :error, "Passwords do not match" }
-  defp validate_password(_), do: {:error, "Password Required"}
+  defp validate_pass(%{other: %{password: ""}}) do
+    {:error, "Password required"}
+  end
+  defp validate_pass(%{other: %{password: pw, password_confirmation: pw}}) do
```
Comment (doomspork) [inline web/models/user_from_auth.ex]: @hassox are you okay with this change?  With `validate_password` and `%Credentials` it was impossible to get under 80.

Comment (hassox) [inline web/models/user_from_auth.ex]: Looks fine to me mate :D

**Pull request merged**